contracts/Comptroller.sol

```diff
@@ -1150,20 +1200,30 @@ contract Comptroller is ComptrollerV5Storage, ComptrollerInterface, ComptrollerE
      */
     function distributeSupplierComp(address cToken, address supplier) internal {
         CompMarketState storage supplyState = compSupplyState[cToken];
-        Double memory supplyIndex = Double({mantissa: supplyState.index});
-        Double memory supplierIndex = Double({mantissa: compSupplierIndex[cToken][supplier]});
-        compSupplierIndex[cToken][supplier] = supplyIndex.mantissa;
+        uint supplyIndex = supplyState.index;
+        uint supplierIndex = compSupplierIndex[cToken][supplier];
+
+        // Update supplier's index to the current index since we are distributing accrued COMP
+        compSupplierIndex[cToken][supplier] = supplyIndex;
 
-        if (supplierIndex.mantissa == 0 && supplyIndex.mantissa > 0) {
-            supplierIndex.mantissa = compInitialIndex;
+        if (supplierIndex == 0 && supplyIndex > 0) {
+            // Covers the case where users supplied tokens before liquidity rewards were a thing
+            // Rewards the user with COMP accrued from the very start of the market
+            supplierIndex = compInitialIndex;
         }
 
-        Double memory deltaIndex = sub_(supplyIndex, supplierIndex);
+        // Calculate change in the cumulative sum of the COMP per cToken accrued
+        Double memory deltaIndex = Double({mantissa: sub_(supplyIndex, supplierIndex)});
```

Comment:
These start to get weird to me bc in one case you have 'index' meaning a mantissa and the other its the whole Double

Reply:
supplyIndex and supplierIndex used to be stored as doubles, and then deltaIndex is the difference between the two.

Since double subtraction is just standard unit (mantissa) subtraction, I made these changes to save on gas. :slightly_smiling_face: